test/assert.js

```diff
@@ -237,3 +283,129 @@ QUnit.test( "raises, alias for throws", function( assert ) {
 		throw "my error";
 	});
 });
+
+QUnit.module( "failing assertions", {
+	beforeEach: function( assert ) {
+		var originalPush = QUnit.assert.push;
+
+		this.originalPush = originalPush;
+
+		QUnit.assert.push = function( result, actual, expected, message ) {
```

Comment:
This should use `assert.push`, that way you're only modifying the current assert instance without requiring access to the global one. May remove the need for the `afterEach` below.